thegame/engine/game_objects/game_object.py

```diff
@@ -1,43 +1,81 @@
 class GameObject:
     """ An object that will be used to fill the objects sheet of a game."""
 
-    def __init__(self, sprite_location: str, animation: str = None, name: str = None):
-
-        self.sprite_location = sprite_location
+    def __init__(
+        self,
+        sprite_locations: dict,
+        initial_sprite: str = None,
+        animation: str = None,
+        name: str = None,
+    ):
+        """ 
+            sprite_locations {name,location}
+            _loaded_sprites {location, Sprite}
+        """
+        self.sprite_locations = sprite_locations
+        self._loaded_sprites = {}
         self.animation = animation
-        self._loaded_sprite = None
         self.name = name
+        if initial_sprite is None and bool(sprite_locations):
+            self.active_sprite = list(self.sprite_locations.keys())[0]
+        else:
+            self.active_sprite = initial_sprite
 
     def clone(self):
         clone = GameObject(
-            sprite_location=self.sprite_location, animation=self.animation
+            sprite_locations=self.sprite_locations,
+            animation=self.animation,
+            initial_sprite=self.active_sprite,
         )
 
         return clone
 
-    def register_loaded_sprite(self, loaded_sprite):
-        self._loaded_sprite = loaded_sprite
+    def register_loaded_sprite(self, loaded_sprite, location):
+        self._loaded_sprites[location] = loaded_sprite
+
+    def deregister_loaded_sprites(self):
+        self.active_sprite = None
+        self._loaded_sprites = {}
 
     def set_sprite_position(self, pos_x, pos_y):
-        if self._loaded_sprite is None:
+        if self._loaded_sprites == {}:
             raise AttributeError(
                 f"Tried to set {type(self).__name__}'s position without registering a loaded sprite."
             )
+        loaded_sprite = self._loaded_sprites[self.sprite_locations[self.active_sprite]]
 
-        self._loaded_sprite.rect = self._loaded_sprite.image.get_rect()
-        self._loaded_sprite.rect.x = pos_x
-        self._loaded_sprite.rect.y = pos_y
+        loaded_sprite.rect = loaded_sprite.image.get_rect()
+        loaded_sprite.rect.x = pos_x
+        loaded_sprite.rect.y = pos_y
+
+    def add_sprite_location(self, name, location):
+        self.sprite_locations[name] = location
 
     def get_sprite(self):
-        if self._loaded_sprite is None:
+        # if self._loaded_sprites is empty
+        if not bool(self._loaded_sprites):
+            raise AttributeError(
+                f"Tried to get {type(self).__name__}'s sprite without registering any sprites."
+            )
+        # if self.active_sprite is None but _loaded_sprites is not empty
+        if self.active_sprite is None:
```

Comment:
Is it possible under some flow that active_sprite won't be in the dictionary but also won't be None?

Reply:
Good point. The only way that would happen is if the initial sprite argument in the init wasn't in the given dictionary. Turns out I missed that edge case there anyways so good call. I've also added the check here in get_sprite() as well, but I think it's redundant. Tell me if you think I should leave it out @Ross-Alexandra

Reply:
I agree, less code over redundant code. If its an error case in the init, then the only way that it'll error out here is if the user went messing with the data (which we can't help, its Python.)

As long as theres no way for it to get through the init in the invalid state, then I don't think we need to check here.

Reply:
Sounds good. Check has been added to init but not get_sprite().